wagtail_live/models.py

```diff
@@ -59,6 +58,168 @@ class LivePageMixin(models.Model):
         StreamFieldPanel("live_posts"),
     ]
 
+    def _get_live_post_index(self, live_post_id, low, high):
+        """ Retrieve index of a live post in page's live posts.
+
+        Uses binary search since live posts are sorted by timestamp.
+        This implementation may change.
+
```

Comment:
Lol, I did this more for the receiving part. When a message is edited/deleted, we have to retrieve it fast, process it and tell the messaging app everything is OK. If you can propose a better alternative for retrieving live posts, I will be happy to implement it.
I tried to use bisect but gave up quickly lol. I liked doing it from scratch but I guess it's best indeed to use built-in functions whenever possible.

So, I will wait for your suggestions before doing more changes for `get_live_post_index`.

Reply:
Ah ok. I think it will not be the bottleneck before you have at least 100.000 messages. I think the saving to the database will be the biggest bottleneck instead. Also if you start at the end and search backwards you will 90% of the time have the message almost immediately, because usually people will edit/delete their most recent messages.

As for bisect, you are right that it doesn't seem to work well when you want to compare by key. It does work when applied like this:

```
class LookUp:
  def __getitem__(self, i):
    return float(self.live_posts[i].id)
    
index = bisect.bisect(LookUp(), float(live_post_id), low, high)
```

But it doesn't look very nice.

Reply:
> I think it will not be the bottleneck before you have at least 100.000 messages

Ah ok. I don't think we'll have to deal often with a live page with at least 100.000 live posts.

>Also if you start at the end and search backwards you will 90% of the time have the message almost immediately, because usually people will edit/delete their most recent messages.

I like this approach!

> I think the saving to the database will be the biggest bottleneck instead.

Is there something we can do in that field?

> As for bisect, you are right that it doesn't seem to work well when you want to compare by key.

Yup!